CMakeLists.txt

```diff
@@ -1,31 +1,20 @@
-cmake_minimum_required(VERSION 2.8.3)
-project(serial)
+cmake_minimum_required(VERSION 3.5.0)
 
-# Find catkin
-find_package(catkin REQUIRED)
+# General setup
+option(BUILD_SHARED_LIBS "Build using shared libraries" ON)
```

Comment:
This is a bit heavy handed. If you're trying to solve the problem of needing `-fPIC`, the idiomatic answer is that users can specify `-DCMAKE_POSITION_INDEPENDENT_CODE=ON` when configuring this project. 

The problem with changing default values is that if anyone includes serial via `add_subdirectory` in their own project, they will inherit this new default value for shared libraries which can be confusing and annoying.

Reply:
The new changes look good to me. GitHub won't let me resolve this comment though.